src/AnnotationReader.php

```diff
@@ -0,0 +1,279 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license. For more information, see
+ * <http://www.doctrine-project.org>.
+ */
+
+declare(strict_types=1);
+
+namespace Doctrine\Annotations;
+
+use ReflectionClass;
+use ReflectionMethod;
+use ReflectionProperty;
+
+use Doctrine\Annotations\Parser\DocParser;
+use Doctrine\Annotations\Parser\MetadataParser;
+use Doctrine\Annotations\Annotation\IgnoreAnnotation;
+
+/**
+ * A reader for docblock annotations.
+ *
+ * @author Jonathan Wage <jonwage@gmail.com>
+ * @author Benjamin Eberlei <kontakt@beberlei.de>
+ * @author Roman Borschel <roman@code-factory.org>
+ * @author Johannes M. Schmitt <schmittjoh@gmail.com>
+ * @author Fabio B. Silva <fabio.bat.silva@hotmail.com>
+ * @author Guilherme Blanco <guilhermeblanco@hotmail.com>
+ */
+class AnnotationReader implements Reader
+{
+    /**
+     * @var \Doctrine\Annotations\Configuration
+     */
+    private $config;
+
+    /**
+     * @var \Doctrine\Annotations\Parser\PhpParser
+     */
+    private $phpParser;
+
+    /**
+     * @var \Doctrine\Annotations\Parser\DocParser
+     */
+    private $docParser;
+
+    /**
+     * @var \Doctrine\Annotations\Parser\MetadataParser
+     */
+    private $metadataParser;
+
+    /**
+     * Constructor.
+     *
+     * @param \Doctrine\Annotations\Configuration $config
+     */
+    public function __construct(Configuration $config = null)
+    {
+        $this->config         = $config ?: new Configuration();
+        $this->phpParser      = $this->config->getPhpParser();
+        $this->docParser      = $this->config->getDocParser();
+        $this->metadataParser = $this->config->getMetadataParser();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getClassAnnotations(ReflectionClass $class) : array
+    {
+        $docblock  = $class->getDocComment();
+        $namespace = $class->getNamespaceName();
+        $imports   = $this->getClassImports($class);
+        $ignored   = $this->getIgnoredAnnotationNames($class);
+        $context   = new Context($class, [$namespace], $imports, $ignored);
+
+        if ($docblock === false) {
```

Comment:
We should prevent all these local variables to be even instantiated if we realize that `$docblock` is empty. Here is a better stub code:

```php
if (($docblock = $class->getDocComment()) === false) {
    return [];
}
```